resolwe_bio/processes/slamdunk/alleyoop_collapse.py

```diff
@@ -61,6 +86,41 @@ def run(self, inputs, outputs):
         collapsed_output = name + '_collapsed.txt'
         os.rename(name + '_collapsed.csv', collapsed_output)
 
+        # normalize to TPM
+        tcount_tpm = compute_tpm(collapsed_output)
+
+        # Map gene symbols to feature IDs
+        res = resdk.Resolwe()
+        CHUNK_SIZE = 1000
+        feature_dict = {}
+        out_columns = [
+            'gene_symbol',
+            'length',
+            'readsCPM',
+            'readsTPM',
+            'conversionRate',
+            'Tcontent',
+            'coverageOnTs',
+            'conversionsOnTs',
+            'readCount',
+            'tcReadCount',
+            'multimapCount',
+        ]
+
+        input_features = tcount_tpm.index.tolist()
+        features_sublists = [input_features[i:i + CHUNK_SIZE] for i in range(0, len(input_features), CHUNK_SIZE)]
+
+        for fsublist in features_sublists:
+            features = res.feature.filter(
+                source=inputs.source,
+                species=inputs.slamdunk.species,
+                feature_id__in=fsublist
+            )
+            feature_dict.update({f.feature_id: f.name for f in features})
```

Comment:
I'm curious what's the output of this if features includes more than one entry? Is this chunk below comparable to what you're trying to do?

```
>>> features = [1, 2, 3]
>>> {'kle': f for f in features}
{'kle': 3}
```

Reply:
Ignore my comment, I just realized that I should have been doing `>>> features = [1, 2, 3]
>>> {'kle': f for f in features}` to make it comparable.

Reply:
Yes, ``f.feature_id`` is unique for a given ``species``/``build`` combination, thus the length of features list is the same as the number of elements in the resulting dict.